test/tap/tests/mysql-protocol_compression_level-t.cpp

```diff
@@ -0,0 +1,184 @@
+	if (!proxysql_compression) {
+		return exit_status();
+	}
+
+	// MySQL connection with compression
+	MYSQL* mysql_compression = initilize_mysql_connection(cl.host, cl.username, cl.password, cl.mysql_port, true);
+	if (!mysql_compression) {
+		return exit_status();
+	}
+
+	// ProxySQL admin connection
+	MYSQL* proxysql_admin = initilize_mysql_connection(cl.host, cl.admin_username, cl.admin_password, cl.admin_port, false);
+	if (!proxysql_admin) {
+		return exit_status();
+	}
+
+	MYSQL_QUERY(proxysql, "CREATE DATABASE IF NOT EXISTS test");
+	MYSQL_QUERY(proxysql, "DROP TABLE IF EXISTS test.sbtest1");
+
+	mysql_query(proxysql, "CREATE TABLE IF NOT EXISTS test.sbtest1 (id INT UNSIGNED NOT NULL AUTO_INCREMENT, k INT UNSIGNED NOT NULL DEFAULT 0, c CHAR(120) NOT NULL DEFAULT '', pad CHAR(60) NOT NULL DEFAULT '', PRIMARY KEY (id), KEY k_1 (k));");
+
+	MYSQL_QUERY(proxysql, "USE test");
+
+	for (int i = 0; i < 100; i++) {
+		MYSQL_QUERY(proxysql, "INSERT INTO sbtest1 (k, c, pad) SELECT FLOOR(RAND() * 10000), REPEAT('a', 120), REPEAT('b', 60) FROM information_schema.tables LIMIT 1000;");
+	}
+
+	std::string query = "SELECT t1.id id1, t1.k k1, t1.c c1, t1.pad pad1, t2.id id2, t2.k k2, t2.c c2, t2.pad pad2 FROM test.sbtest1 t1 JOIN test.sbtest1 t2 LIMIT 90000000";
+
+	unsigned long time_proxy = calculate_query_execution_time(proxysql, query);
+	diag("Time taken for query with proxysql without compression: %ld", time_proxy);
+
+	unsigned long time_proxy_compressed = calculate_query_execution_time(proxysql_compression, query);
+	diag("Time taken for query with proxysql with compression: %ld", time_proxy_compressed);
+
+	unsigned long diff = abs(time_proxy - time_proxy_compressed);
+	int performance_diff = (diff * 100) / time_proxy;
+
+	ok((performance_diff < 10), "proxysql with compression performed well compared to without compression. Performance difference: %d percentage", performance_diff);
+
+	unsigned long time_mysql_compressed = calculate_query_execution_time(mysql_compression, query);
+	diag("Time taken for query with mysql with compression: %ld", time_mysql_compressed);
+
+	diff = abs(time_mysql_compressed - time_proxy_compressed);
+	performance_diff = (diff * 100) / time_mysql_compressed;
+
+	ok((performance_diff < 20), "proxysql with compression performed well compared to mysql with compression. Performance difference: %d percentage", performance_diff);
+
+	int32_t compression_level = get_compression_level(proxysql_admin);
+	ok(compression_level == 3, "Default compression level is correct: %d", compression_level);
+
+	set_compression_level(proxysql_admin, "8");
+	compression_level = get_compression_level(proxysql_admin);
+	ok(compression_level == 8, "Compression level set correctly: %d", compression_level);
+
+	time_proxy_compressed = calculate_query_execution_time(proxysql_compression, query);
+	diag("Time taken for query with proxysql with compression level 8: %ld", time_proxy_compressed);
+
+	set_compression_level(proxysql_admin, "3");
```

Comment:
Individual value check isn't bad, but since the range is small, a simple `for` here can be used to verify the whole range, or at least check the range boundaries, checking the failures outside the range. Also, the table `runtime_global_variables` is never checked, it's possible for a variable to be set, but latter ProxySQL refusing to load it to runtime, since boundaries checks are enforced when `LOAD TO RUNTIME` is issued to `Admin` interface. When verifying if any config table is properly changed, it's important to check the table, but also it's `runtime_*` counterpart.

Reply:
I see that in the new logic only runtime is checked, it should be fine, but here it was suggested to check both, just wondering, any reason for it? Thanks.

Reply:
No specfic reason, added Thanks!